### ISSUE TITLE
Versions author info is lost after delete and restore by share receiver

### ISSUE THREAD

**Issue by jnweiger** [owncloud/core#39571]
Seen in 10.9.0 beta1

* user01 creates a file xxx.txt and shares it with admin.
* add `'file_storage.save_version_author' => true,` to config.php
* both user01 and admin edit the file, to create multiple versions.
* author version information is shown correctly as implemented in #39126 . OK.
* admin deletes the file.
* The trashcan has copies of the versioned file xxx.txt for both user01 and admin: (OK)
```
# ls data/user01/files_trashbin/versions/xxx.txt.v163892
xxx.txt.v1638928947.d1638929506       xxx.txt.v1638929221.d1638929506       xxx.txt.v1638929228.json.d1638929506  xxx.txt.v1638929442.d1638929506
xxx.txt.v1638928975.d1638929506       xxx.txt.v1638929221.json.d1638929506  xxx.txt.v1638929230.d1638929506       xxx.txt.v1638929442.json.d1638929506
xxx.txt.v1638928976.d1638929506       xxx.txt.v1638929222.d1638929506       xxx.txt.v1638929230.json.d1638929506  xxx.txt.v1638929454.d1638929506
xxx.txt.v1638928989.d1638929506       xxx.txt.v1638929222.json.d1638929506  xxx.txt.v1638929436.d1638929506       xxx.txt.v1638929454.json.d1638929506
xxx.txt.v1638928989.json.d1638929506  xxx.txt.v1638929223.d1638929506       xxx.txt.v1638929436.json.d1638929506  xxx.txt.v1638929455.d1638929506
xxx.txt.v1638929156.d1638929506       xxx.txt.v1638929223.json.d1638929506  xxx.txt.v1638929437.d1638929506       xxx.txt.v1638929455.json.d1638929506
xxx.txt.v1638929156.json.d1638929506  xxx.txt.v1638929224.d1638929506       xxx.txt.v1638929437.json.d1638929506  xxx.txt.v1638929456.d1638929506
xxx.txt.v1638929158.d1638929506       xxx.txt.v1638929224.json.d1638929506  xxx.txt.v1638929438.d1638929506       xxx.txt.v1638929456.json.d1638929506
xxx.txt.v1638929158.json.d1638929506  xxx.txt.v1638929225.d1638929506       xxx.txt.v1638929438.json.d1638929506  xxx.txt.v1638929457.d1638929506
xxx.txt.v1638929159.d1638929506       xxx.txt.v1638929225.json.d1638929506  xxx.txt.v1638929439.d1638929506       xxx.txt.v1638929457.json.d1638929506
xxx.txt.v1638929159.json.d1638929506  xxx.txt.v1638929226.d1638929506       xxx.txt.v1638929439.json.d1638929506  xxx.txt.v1638929458.d1638929506
xxx.txt.v1638929160.d1638929506       xxx.txt.v1638929226.json.d1638929506  xxx.txt.v1638929440.d1638929506       xxx.txt.v1638929458.json.d1638929506
xxx.txt.v1638929160.json.d1638929506  xxx.txt.v1638929227.d1638929506       xxx.txt.v1638929440.json.d1638929506  xxx.txt.v1638929459.d1638929506
xxx.txt.v1638929162.d1638929506       xxx.txt.v1638929227.json.d1638929506  xxx.txt.v1638929441.d1638929506       xxx.txt.v1638929459.json.d1638929506
xxx.txt.v1638929162.json.d1638929506  xxx.txt.v1638929228.d1638929506       xxx.txt.v1638929441.json.d1638929506  

# ls data/admin/files_trashbin/versions/xxx.txt.v163892
xxx.txt.v1638928947.d1638929506  xxx.txt.v1638929162.d1638929506  xxx.txt.v1638929228.d1638929506  xxx.txt.v1638929442.d1638929506
xxx.txt.v1638928975.d1638929506  xxx.txt.v1638929221.d1638929506  xxx.txt.v1638929230.d1638929506  xxx.txt.v1638929454.d1638929506
xxx.txt.v1638928976.d1638929506  xxx.txt.v1638929222.d1638929506  xxx.txt.v1638929436.d1638929506  xxx.txt.v1638929455.d1638929506
xxx.txt.v1638928989.d1638929506  xxx.txt.v1638929223.d1638929506  xxx.txt.v1638929437.d1638929506  xxx.txt.v1638929456.d1638929506
xxx.txt.v1638929156.d1638929506  xxx.txt.v1638929224.d1638929506  xxx.txt.v1638929438.d1638929506  xxx.txt.v1638929457.d1638929506
xxx.txt.v1638929158.d1638929506  xxx.txt.v1638929225.d1638929506  xxx.txt.v1638929439.d1638929506  xxx.txt.v1638929458.d1638929506
xxx.txt.v1638929159.d1638929506  xxx.txt.v1638929226.d1638929506  xxx.txt.v1638929440.d1638929506  xxx.txt.v1638929459.d1638929506
xxx.txt.v1638929160.d1638929506  xxx.txt.v1638929227.d1638929506  xxx.txt.v1638929441.d1638929506  
```

The copies saved for the share receiver admin does not duplicate the json files. BAD.
When admin restores the file, the version author information is lost. BAD.

Expected behaviour:
* all trashbin copies replicate not only the versions but also the .json meta data for the versions.
* author information is retained across delete/restore

**Comment by jnweiger** [comment]
@IljaN this is your's, I guess :-)

**Comment by phil-davis** [comment]
I will make some test scenarios that demonstrate this (and other workflows that I think will be useful and interesting to test). That should help us make sure to fix the various combinations of this beast, without accidentally regressing.

**Comment by JammingBen** [comment]
Should be fixed with the latest commit to https://github.com/owncloud/core/pull/39516.

**Comment by JammingBen** [comment]
Closing there as it should be fixed. Just re-open if that's not the case :)

**Comment by jnweiger** [comment]
Confirmed fixed in 10.9.0 rc1. Thanks!
